tests/test_mkl_service.py

```diff
@@ -24,317 +24,306 @@
+@pytest.mark.parametrize('branch', branches + strict)
+def test_cbwr_all(branch):
+    check_cbwr(branch, 'all')
 
-    def test_enable_instructions_avx512(self):
-        mkl.enable_instructions('avx512')
+def check_cbwr(branch, cnr_const):
+    status = mkl.cbwr_set(branch=branch)
+    if status == 'success':
+        expected_value = 'branch_off' if branch == 'off' else branch
+        actual_value = mkl.cbwr_get(cnr_const=cnr_const)
+        assert actual_value == expected_value, \
+            f"Round-trip failure for CNR branch '{branch}', CNR const '{cnr_const}"
+    elif status != 'err_unsupported_branch':
+        pytest.fail(status)
 
-    def test_enable_instructions_avx512_mic(self):
-        mkl.enable_instructions('avx512_mic')
+def test_cbwr_get_auto_branch():
+    mkl.cbwr_get_auto_branch()
 
-    def test_enable_instructions_avx2(self):
-        mkl.enable_instructions('avx2')
 
-    def test_enable_instructions_avx(self):
-        mkl.enable_instructions('avx')
+def test_enable_instructions_avx512_mic_e1():
+    mkl.enable_instructions('avx512_mic_e1')
 
-    def test_enable_instructions_sse4_2(self):
-        mkl.enable_instructions('sse4_2')
+def test_enable_instructions_avx512():
+    mkl.enable_instructions('avx512')
 
-    def test_set_env_mode(self):
-        mkl.set_env_mode()
+def test_enable_instructions_avx512_mic():
+    mkl.enable_instructions('avx512_mic')
 
-    def test_get_env_mode(self):
-        mkl.get_env_mode()
+def test_enable_instructions_avx2():
+    mkl.enable_instructions('avx2')
 
-    def test_verbose_false(self):
-        mkl.verbose(False)
+def test_enable_instructions_avx():
+    mkl.enable_instructions('avx')
 
-    def test_verbose_true(self):
-        mkl.verbose(True)
+def test_enable_instructions_sse4_2():
+    mkl.enable_instructions('sse4_2')
 
-    #def test_set_mpi_custom(self):
-    #    mkl.set_mpi('custom', 'custom_library_name')
+def test_set_env_mode():
+    mkl.set_env_mode()
 
-    @skip
-    def test_set_mpi_msmpi(self):
-        mkl.set_mpi('msmpi')
+def test_get_env_mode():
+    mkl.get_env_mode()
 
-    #def test_set_mpi_intelmpi(self):
-    #    mkl.set_mpi('intelmpi')
+def test_verbose_false():
+    mkl.verbose(False)
 
-    #def test_set_mpi_mpich2(self):
-    #    mkl.set_mpi('mpich2')
+def test_verbose_true():
+    mkl.verbose(True)
 
+#def test_set_mpi_custom():
+#    mkl.set_mpi('custom', 'custom_library_name')
 
-class test_vm_service_functions(unittest.TestCase):
-    def test_vml_set_get_mode_roundtrip(self):
-        saved = mkl.vml_get_mode()
-        mkl.vml_set_mode(*saved) # should not raise errors
+@pytest.mark.skip
+def test_set_mpi_msmpi():
+    mkl.set_mpi('msmpi')
 
-    def test_vml_set_mode_ha_on_ignore(self):
-        mkl.vml_set_mode('ha', 'on', 'ignore')
+#def test_set_mpi_intelmpi():
+#    mkl.set_mpi('intelmpi')
 
-    def test_vml_set_mode_ha_on_errno(self):
-        mkl.vml_set_mode('ha', 'on', 'errno')
+#def test_set_mpi_mpich2():
```

Comment:
Please use `pytest.marks.skip` decorator to skip this test.

Reply:
Added